client/app/components/ParameterMappingInput.jsx

```diff
@@ -240,6 +311,41 @@ export class ParameterMappingListInput extends React.Component {
     Query: null,
   };
 
+  constructor(props) {
+    super(props);
+    this.wrapperRef = React.createRef();
+  }
+
+  static getStringValue(value) {
+    // null
+    if (!value) {
+      return '';
+    }
+
+    // array
+    if (value instanceof Array) {
+      const arr = value.map(v => this.getStringValue(v));
+      const delimiter = moment.isMoment(value[0]) ? ' ~ ' : ', ';
+      return arr.join(delimiter);
+    }
+
+    // moment
+    if (moment.isMoment(value)) {
+      return value.format('DD/MM/YY');
+    }
+
+    // rest
+    return value.toString();
+  }
+
+  static getDefaultValue(mapping) {
+    const value = mapping.type === MappingType.StaticValue
+      ? mapping.value || mapping.param.normalizedValue
+      : mapping.param.normalizedValue;
+
+    return this.getStringValue(value);
```

Comment:
Enum, string, date, date range, number
<img width="197" alt="screen shot 2019-01-23 at 21 49 48" src="https://user-images.githubusercontent.com/486954/51632829-df924900-1f58-11e9-8e69-6e3e04f065b0.png">

I made it so _all_ date types would format to DD/MM/YY. Lmk if you think it needs refining.

Reply:
You should use format from `clientConfig.dateFormat`:
https://github.com/getredash/redash/blob/master/client/app/components/DateInput.jsx#L14
https://github.com/getredash/redash/blob/master/client/app/components/DateTimeInput.jsx#L15-L16

Reply:
@kravets-levko I found a way around it, leveraging the already formatted `param.value` instead of the raw `param.normalizedValue`. So now, not only is the date correctly formatted, but also shows time and seconds when relevant.

https://github.com/getredash/redash/pull/3332/commits/48743c7eda6da7aa15d1698ee30d69e3bc4c80e5

Reply:
Please make sure that you're using `clientConfig.dateFormat` only when showing it to the user. When setting the value for the parameter you need to use the static format we were using (`YYYY-MM-DD HH:mm:ss`).

Reply:
In 48743c7 there's no formatting being done. Just taking the already formatted `mapping.param.value`.

Reply:
Damn, there's a case where static value is not yet normalized and it turns to shit. I hope I don't have to resort to dupe-formatting 😭

Reply:
This has been resolved :)